### PR TITLE
Desktop View: Home Page & Navigation Bar

### PR DESCRIPTION
**Pull request by blakeheyman**
This PR adds desktop styling to the homepage of the website. This includes the card layout, header text, and navigation bar.

### Changes:
- Rework flex layout on cards to support desktop using media queries
- Move the "Home" label above/below the image using media queries
- Added NavBar component
  - Dropdown support for quick links
- Added MenuBar component
  - Wraps around NavBar and SideBar to display the appropriate one depending on the window size.

### TODO
- Update navigation bar icons once designs are finalized
- Support routing for the headers themselves in navigation bar

### Known Bugs
- Dropdown menu is layered behind the image on the homepage

### Testing
No automated testing has been done. The changes can be validated through `npm run start-frontend` and resizing the window to desktop size.

### Screens

| Mobile | Desktop |
|----|----|
| HomepageMobile | HomepageDesktop |

### CODE REVIEW

File: frontend/components/Layout.tsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
-import Sidebar from './Sidebar';
+import MenuBar from './Menubar';
 import styled from 'styled-components';
 import { ReactElement } from 'react';
 
 const PageContainer = styled.div`
   width: 100vw;
```
Comment (stefan-philip) [inline frontend/components/Layout.tsx:6]: maybe make this 100% ?

Getting some weird behavior where when I extend my screen width, the background doesn't fill in

Screen Shot 2022-02-15 at 11 52 11 PM

**Review by stefan-philip — CHANGES_REQUESTED** [review]
looks great guys

one comment that I'd like to address and I couldn't pinpoint
when we extend the screen's width (in mobile view), I don't want to have to scroll down to see more content
see if you can think of some way to play around with the sizing of the image (I think) so that we dont have to scroll when extending width

File: frontend/components/Layout.tsx [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
   height: 100vh;
-  position: relative;
+  background-color: #ef8b6f;
```
[the remark this answers is not in the document]
Comment (blakeheyman) [reply frontend/components/Layout.tsx]: Added in `mediaSizes.tsx`

File: frontend/components/NavBar.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  margin-top: 45px;
+  margin-left: 40px;
+`;
+
+const MenuHeading = ({
+  title,
+  links = [],
+}: {
+  title: string;
+  links?: Link[];
```
[the remark this answers is not in the document]
Comment (blakeheyman) [reply frontend/components/NavBar.tsx:73]: With the current implementation, you don't have to specify links if you don't need a dropdown.

File: frontend/pages/home.tsx [hunk 4]
```diff
@@ -4,39 +4,58 @@ import styled from 'styled-components';
...
 
 const HomeContainer = styled.div`
-  background-color: #ef8b6f;
   display: flex;
   flex-direction: column;
-  min-height: 100vh;
   padding: 2rem;
+  background-color: ${colors.homepageBackground};
+  .title-below {
+    #mi
```
Comment (stefan-philip) [inline frontend/pages/home.tsx:15]: #mi ?

File: frontend/pages/home.tsx [hunk 5]
```diff
@@ -4,39 +4,58 @@ import styled from 'styled-components';
...
+    @media min-width: ${size.laptop + 1}px {
+      display: block;
+    }
+  }
```
Comment (stefan-philip) [inline frontend/pages/home.tsx:19]: still noticing odd behavior at the breakpoint
title_above appears right at the breakpoint, and disappears when its larger screen

File: frontend/pages/home.tsx [hunk 6]
```diff
@@ -4,39 +4,58 @@ import styled from 'styled-components';
...
 `;
 
 const ImageContainer = styled(Row)`
   margin: 1rem 0rem 1rem 0rem;
+  @media ${maxDevice.laptop} {
+    height: auto;
+    max-width: 600px;
+  }
 `;
+const ImageContent = styled(Image)``;
```
Comment (stefan-philip) [inline frontend/pages/home.tsx:48]: kinda bothers me that we have to scroll vertically when we enter "tablet" size territory
could we set a max-height, probably based on `vh` or `vw`

File: frontend/components/Card.tsx [hunk 7]
```diff
@@ -12,7 +13,25 @@ const StyledCard = styled(AntdCard)`
   flex-direction: column;
   font-size: 0.8rem;
   padding: 1rem;
-  margin: 0.25rem;
+  width: 30%;
+  max-width: 281px;
```
Comment (stefan-philip) [inline frontend/components/Card.tsx:17]: can remove this max-width line, as it gets overridden in the media queries

File: frontend/components/Card.tsx [hunk 8]
```diff
@@ -12,7 +13,25 @@ const StyledCard = styled(AntdCard)`
...
+    flex-grow: 1;
+    width: 45%;
+    max-width: 100%;
+    margin: 0.25rem;
+    height: 138px;
+  }
+
+  @media (min-width: ${size.laptop}px) {
+    &:nth-child(3) {
+      max-width: 381px;
```
Comment (stefan-philip) [inline frontend/components/Card.tsx:30]: do this in terms of `vh` or `vw`

File: frontend/components/Card.tsx [hunk 9]
```diff
@@ -12,7 +13,25 @@ const StyledCard = styled(AntdCard)`
...
+  width: 30%;
+  max-width: 281px;
+  margin-right: min(3%, 39px);
+
+  @media ${maxDevice.laptop} {
+    flex-grow: 1;
+    width: 45%;
+    max-width: 100%;
+    margin: 0.25rem;
+    height: 138px;
```
Comment (stefan-philip) [inline frontend/components/Card.tsx:25]: make it related to `vh` for responsiveness

File: frontend/components/Menubar.tsx [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+const Menubar = (): ReactElement => {
+  const windowSize = useWindowSize();
+
+  if (!windowSize.width) return <Sidebar />;
+
+  return windowSize.width < size.laptop ? <Sidebar /> : <NavBar />;
+};
+
+// Hook - https://usehooks.com/useWindowSize/
+function useWindowSize() {
```
Comment (stefan-philip) [inline frontend/components/Menubar.tsx:15]: awesome idea. lets place this function inside a `hooks` directory though, just in case we want to reuse it

**Review by stefan-philip — CHANGES_REQUESTED** [review]

File: frontend/pages/home.tsx (hunk 4 above)
Comment (DilanRocco) [reply frontend/pages/home.tsx:15]: typo

File: frontend/components/Card.tsx (hunk 7 above)
Comment (DilanRocco) [reply frontend/components/Card.tsx:17]: Screen Shot 2022-02-22 at 11 28 38 AM

If I was to remove this Stefan, the cards in the navbar mode would grow forever like shown in this picture.

File: frontend/components/NavBar.tsx [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+}): ReactElement => {
+  const [dropdown, setDropdown] = useState(false);
+
+  const linkToItem = (link: Link) => {
+    return (
+      <MenuItem>
+        <a target="_blank" rel="noopener noreferrer" href={link.url}>
```
Comment (stefan-philip) [inline frontend/components/NavBar.tsx:80]: make this a next/Link component

File: frontend/pages/home.tsx (hunk 6 above)
Comment (stefan-philip) [reply frontend/pages/home.tsx:47]: here's what I'm seeing:

when I'm in mobile view, its fine, no need to scroll vertically:
Screen Shot 2022-02-22 at 1 42 58 PM

when I extend the window width, the image gets larger, so I have to scroll vertically:
Screen Shot 2022-02-22 at 1 43 11 PM

ideally, there's some solution that allows me to extend the width of my screen, without having to scroll down to see the bottom of the page

File: frontend/components/Card.tsx (hunk 9 above)
Comment (stefan-philip) [reply frontend/components/Card.tsx:25]: actually, lets get rid of the line `height: 138px;` 
that'll make the cards more responsive (not tied to a pixel value)

File: frontend/components/Card.tsx [hunk 12]
```diff
@@ -12,7 +13,27 @@ const StyledCard = styled(AntdCard)`
...
+
+  @media (min-width: ${size.laptop + 1}px) {
+    &:nth-child(3) {
+      max-width: 41vh;
+      width: 30%;
+    }
+
+    max-width: 31vh;
+    padding: 2rem;
+  }
```
Comment (stefan-philip) [inline frontend/components/Card.tsx:36]: Played around with the Card styling, here's what I like:


styled(AntdCard)`
   background-color: white;
   flex-direction: column;
   font-size: 0.8rem;
   padding: 1rem;

   margin-right: 3%;

   @media ${maxDevice.laptop} {
     flex-grow: 1;
     width: 45%;
     max-width: 100%;
     margin: 0.25rem;
  }
  @media (min-width: ${size.laptop + 1}px) {
     &:nth-child(3) {
       max-width: 35%;
       margin-right: 0;
     }

     max-width: 28%
     padding: 2rem;
   }

File: frontend/components/NavBar.tsx [hunk 13]
```diff
@@ -109,7 +109,7 @@ const MenuElements = styled.div`
 `;
 
 const NavMenu = styled.div`
-  height: 120px;
+  height: 11vh;
```
[the remark this answers is not in the document]
Comment (stefan-philip) [reply frontend/components/NavBar.tsx:112]: 85px looks good to me

File: frontend/pages/home.tsx (hunk 6 above)
Comment (stefan-philip) [reply frontend/pages/home.tsx:44]: fixed!

**Review by stefan-philip — APPROVED** [review]

**Pull request merged**